src/__tests__/generators.test.ts

```diff
@@ -0,0 +1,31 @@
+import { idGenerators } from '../generators';
+
+describe('idGenerators', () => {
```

Comment:
I wouldn't recommend testing it this way.
For instance, in this case, the IDs wouldn't be unique because the ctx does not have correct IDs.
You can toss in a spec here src/oas3/__tests__/__fixtures__

Reply:
I am able to confirm the ids are unique in the traverse callback tests I added, so these tests were mainly added to confirm string output from the idGenerator was sanitizing the path still for http_operations and not for http_callbacks. 

The way I understood the id generator to be working was the string from the idGenerator is used as part of generating the actual id on the TransformerContext. So if the string value is unique, the generated id should be unique because its what gets hashed. Is this accurate?

Reply:
>So if the string value is unique, the generated id should be unique because its what gets hashed.

the problem is we don't have a test that'd verify whether the `parentId` actually gets set, and if `parentId` is not set properly, the IDs may no longer be stable if one of the dependencies is always set to `undefined`.
That's why I suggested adding a document do that ids directory.